src/utils/confirmations.ts

```diff
@@ -0,0 +1,98 @@
+ *
+ * @param title - The title of the confirmation dialog.
+ * @param message - The message to display in the confirmation dialog. This message will be suffixed with instructions for a specific prompt.
+ * @param expectedConfirmationWord - The word that the user must type to confirm the action when the confirmation style is set to 'Word Confirmation'.
+ * @returns A promise that resolves to a boolean indicating whether the user confirmed the action.
+ */
+export async function getConfirmationAsInSettings(
+    title: string,
+    message: string,
+    expectedConfirmationWord: string,
+): Promise<boolean> {
+    const deleteConfirmation: string | undefined = vscode.workspace
+        .getConfiguration()
+        .get<string>(ext.settingsKeys.confirmationStyle);
+
+    if (deleteConfirmation === 'Word Confirmation') {
+        return await getConfirmationWithWordQuestion(title, message, expectedConfirmationWord);
+    }
+    return await getConfirmationWithClick(title, message);
+}
+
+export async function getConfirmationWithWordQuestion(
+    title: string,
+    message: string,
+    expectedConfirmationWord: string,
+): Promise<boolean> {
+    const result = await vscode.window.showInputBox({
+        title: title,
+        prompt: `${message}\n\nPlease enter the word "${expectedConfirmationWord}" to confirm the operation.`,
+        ignoreFocusOut: true,
+        validateInput: (val: string | undefined) => {
+            if (val && 0 === val.localeCompare(expectedConfirmationWord, undefined, { sensitivity: 'accent' })) {
+                return undefined;
+            }
+            return `Please enter the word "${expectedConfirmationWord}" to confirm the operation.`;
+        },
+    });
+
+    if (!result) {
+        throw new UserCancelledError();
+    }
+
+    return 0 === result.localeCompare(expectedConfirmationWord, undefined, { sensitivity: 'accent' });
+}
+
+export async function getConfirmationWithNumberQuiz(title: string, message: string): Promise<boolean> {
+    const randomInput: { numbers: number[]; index: number } = getRandomArrayAndIndex(3);
+
+    const confirmation = await vscode.window.showWarningMessage(
+        title,
+        {
+            modal: true,
+            detail:
+                message +
+                `\n\n` +
+                `Pick '${randomInput.numbers[randomInput.index]}' to confirm and continue.\n\n` +
+                `This verification step can be adjusted in Settings.`,
+        },
+        randomInput.numbers[0].toString(),
+        randomInput.numbers[1].toString(),
+        randomInput.numbers[2].toString(),
+    );
+
+    return confirmation === randomInput.numbers[randomInput.index].toString();
+}
+
+export async function getConfirmationWithClick(title: string, message: string): Promise<boolean> {
+    const confirmation = await vscode.window.showWarningMessage(
+        title,
+        {
+            modal: true,
+            detail: message,
+        },
+        'Yes',
```

Comment:
`Yes` will be the default button in this case, I'm still not convinced that a destructive operation should be the default dialog action. It makes it too easy to confirm. We should define our own `No` or `Cancel` button and make it the default choice, so that at least one additional "TAB" would be needed to confirm with the keyboard.

Reply:
@sevoku that's a tricky one, when going for a modal dialog, we automatically get a "Cancel" button provided. The "Cancel" button doesn't get the default flag assigned.. So in the end, the only way to get there.. it's another QuickPick and not a dialog.

I'll push the other changes and will put this one on hold after the blocking ones are fixed.

Reply:
Oh I see now what's going on, since the cancel button has always `isCloseAffordance: true` it makes the other "Yes" button the default automatically. This can only be avoided by either adding more buttons or using a custom Cancel button with `isCloseAffordance: false`, however that would break ESC key behaviour. Probably not much we can do here then. However for the quiz button, consider making the "right" answer not the first button in the list.